File: src/main/java/com/nerdearla/workshop/controller/advice/GlobalControllerAdvice.java
package com.nerdearla.workshop.controller.advice;

import com.nerdearla.workshop.dto.error.APIError;
import org.springframework.http.HttpStatus;
import org.springframework.http.converter.HttpMessageNotReadableException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalControllerAdvice {

    @ExceptionHandler(HttpMessageNotReadableException.class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    public APIError handleException(Exception e) {
        return new APIError(e.getMessage());
    }
}

File: src/main/java/com/nerdearla/workshop/controller/advice/GlobalControllerAdvice.kt
package com.nerdearla.workshop.controller.advice

import com.nerdearla.workshop.dto.error.APIError
import org.springframework.http.HttpStatus
import org.springframework.http.converter.HttpMessageNotReadableException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalControllerAdvice {
    @ExceptionHandler(HttpMessageNotReadableException::class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    fun handleException(e: Exception): APIError {
        return APIError(e.message)
    }
}